crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -1190,6 +1194,44 @@ mod private {
             Ok(updates_as_vector_diffs)
         }
 
+        /// If the event has a thread root, ensure the thread root event has a thread summary.
+        #[instrument(skip_all)]
+        async fn extract_thread_root(&mut self, event: &Event) -> Result<(), EventCacheError> {
+            let Some(thread_root) = extract_thread_root(event.raw()) else {
+                // No thread root, carry on.
+                return Ok(());
+            };
+
+            // Add a thread summary to the event which has the thread root, if we knew about it.
+            let Some((location, mut target_event)) = self.find_event(&thread_root).await? else {
+                trace!("thread root event is missing from the linked chunk");
+                return Ok(());
+            };
+
+            match target_event.thread_summary {
+                ThreadSummaryStatus::Unknown | ThreadSummaryStatus::None => {}
+                ThreadSummaryStatus::Some(_thread_summary) => {
+                    // The event already had a thread summary; ignore.
+                    // TODO: later, we might want to recompute the thread summary here, and cause
+                    // an update if it's changed.
+                    return Ok(());
+                }
+            }
+
+            target_event.thread_summary = ThreadSummaryStatus::Some(ThreadSummary {});
+
+            match location {
+                EventLocation::Memory(position) => {
+                    self.events
+                        .replace_event_at(position, target_event)
+                        .expect("should have been a valid position of an item");
+                }
+                EventLocation::Store => self.save_event([target_event]).await?,
+            }
```

Comment:
I don't understand what it does. The method is extracting the thread root and is saving it?

Also, maybe we could put this `EventLocation` logic inside `save_event`, what do you think?